modules/biobank/php/container.class.inc

```diff
@@ -0,0 +1,339 @@
+     *
+     * @return void
+     */
+    public function setStatusId(int $statusId) : void
+    {
+        $this->statusId = $statusId;
+    }
+
+    /**
+     * Gets the ID of the container's current status
+     *
+     * @return int
+     */
+    public function getStatusId() : int
+    {
+        return $this->statusId;
+    }
+
+    /**
+     * Sets the ID of the container's origin
+     *
+     * @param int $originId the ID of the container's origin
+     *
+     * @return void
+     */
+    public function setOriginId(int $originId) : void
+    {
+        $this->originId = $originId;
+    }
+
+    /**
+     * Gets the ID of the container's origin
+     *
+     * @return int
+     */
+    public function getOriginId() : int
+    {
+        return $this->originId;
+    }
+
+    /**
+     * Sets the ID of the container's current centerId
+     *
+     * @param int $centerId the ID of the container's current centerId
+     *
+     * @return none
+     */
+    public function setCenterId(int $centerId) : void
+    {
+        $this->centerId = $centerId;
+    }
+
+    /**
+     * Gets the ID of the container's current centerId
+     *
+     * @return int
+     */
+    public function getCenterID() : int
+    {
+        return $this->centerId;
+    }
+
+    /**
+     * Sets the parent container's ID
+     *
+     * @param int $parentContainerId the parent container's ID
+     *
+     * @return void
+     */
+    public function setParentContainerId(?int $parentContainerId = null) : void
+    {
+        $this->parentContainerId = $parentContainerId;
+    }
+
+    /**
+     * Gets the parent container's ID
+     *
+     * @return int or void
+     */
+    public function getParentContainerId() : ?int
+    {
+        return $this->parentContainerId;
+    }
+
+    /**
+     * Sets the IDs of the children containers
+     *
+     * @param array $childContainerIds array of IDs of children container
+     *
+     * @return void
+     */
+    public function setChildContainerIds(?array $childContainerIds = null) : void
+    {
+        $this->childContainerIds = $childContainerIds;
+    }
+
+    /**
+     * Gets the IDs of the children containers
+     *
+     * @return array
+     */
+    public function getChildContainerIds() : ?array
+    {
+        return $this->childContainerIds;
+    }
+
+    /**
+     * Sets the container's current coordinate in storage
+     *
+     * @param int $coordinate the container's current coordinate
+     *
+     * @return void
+     */
+    public function setCoordinate(?int $coordinate = null) : void
+    {
+        $this->coordinate = $coordinate;
+    }
+
+    /**
+     * Gets the container's current coordinate in storage
+     *
+     * @return int
+     */
+    public function getCoordinate() : ?int
+    {
+        return $this->coordinate;
+    }
+
+    /**
+     * Gets an array representing the state of the Container
+     *
+     * @return array
+     */
+    public function jsonSerialize() : array
```

Comment:
This doesn't serialize to JSON, just creates an associative array.

Why not use the mandatory toJSON() to do this instead, since you can't get rid of it? Also make sure to use `json_encode` if you are actually using this as JSON.

Reply:
So this function is triggered as soon `json_encode()` is called on this object. It's a feature of implementing `\JsonSerializable`.

http://php.net/manual/en/jsonserializable.jsonserialize.php